airflow/bin/cli.py

```diff
@@ -607,6 +636,10 @@ class CLIFactory(object):
         'execution_date': Arg(
             ("execution_date",), help="The execution date of the DAG",
             type=parsedate),
+        'execution_date_enhanced': Arg(
```

Comment:
@mistercrunch 
I see execution date used in 4 CLI commands currently : `render`, `run`, `test`, and `task_state`. `task_state` seems possible, but I'm not familiar with how `run` is used internally. Also, what does `render` do? I can possibly look at these providing uniform handling if you think this feature makes sense for all of the commands above.